homeassistant/components/vegehub/__init__.py

```diff
@@ -0,0 +1,160 @@
+"""The Vegetronix VegeHub integration."""
+
+from collections.abc import Awaitable, Callable
+from http import HTTPStatus
+import logging
+from typing import Any
+
+from aiohttp.hdrs import METH_POST
+from aiohttp.web import Request, Response
+
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.components.webhook import (
+    async_register as webhook_register,
+    async_unregister as webhook_unregister,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_WEBHOOK_ID, EVENT_HOMEASSISTANT_STOP
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_validation as cv, device_registry as dr
+from homeassistant.helpers.typing import ConfigType
+
+from .const import DOMAIN, NAME, PLATFORMS
+
+_LOGGER = logging.getLogger(__name__)
+
+# The integration is only set up through the UI (config flow)
+CONFIG_SCHEMA = cv.config_entry_only_config_schema(DOMAIN)
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the VegeHub integration."""
+    return True  # For now, we are not using YAML config.
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up VegeHub from a config entry."""
+
+    # Register the device in the device registry
+    device_registry = dr.async_get(hass)
+
+    hass.data.setdefault(DOMAIN, {})
+    device_mac = str(entry.data.get("mac_address"))
+
+    if entry.unique_id is None:
+        new_data = entry.data.copy()
+        unique_id = device_mac
+        hass.config_entries.async_update_entry(
+            entry, data=new_data, unique_id=unique_id
+        )
```

Comment:
you can just do an `assert entry.unique_id` here. It's a new integration and every entry will have a unique_id